libs/community/langchain_community/document_loaders/sharepoint.py

```diff
@@ -82,6 +82,7 @@ def lazy_load(self) -> Iterator[Document]:
                     auth_identities = self.authorized_identities(file_id)
                 if self.load_extended_metadata is True:
                     extended_metadata = self.get_extended_metadata(file_id)
+                    extended_metadata.update({"source_full_url": target_folder.web_url})
```

Comment:
How do you handle this when `folder_id` or `object_ids` is passed instead of `folder_path`, or when none of these is passed (i.e., `root` by default)?

Reply:
Added for `folder_id` and when None is passed i.e., `root` by default.
Not yet for `object_ids`